src/material-examples/dialog-overview/dialog-overview-example.ts

```diff
@@ -22,10 +22,11 @@ export class DialogOverviewExample {
   constructor(public dialog: MatDialog) {}
 
   openDialog(): void {
-    const dialogRef = this.dialog.open(DialogOverviewExampleDialog, {
-      width: '250px',
-      data: {name: this.name, animal: this.animal}
-    });
+    const dialogRef: MatDialogRef<DialogOverviewExampleDialog> =
```

Comment:
@jelbourn based on the ticket description the return type was supposed to be changed to `{}`, but I'm not completely sure that it makes sense. E.g. in this case the `{}` didn't allow me to type the value to `MatDialogRef<DialogOverviewExampleDialog, string>`, because a `string` isn't assignable to `{}`.

Reply:
[AFAIK all primitives are assignable to `{}`](https://www.typescriptlang.org/play/#src=let%20x%3A%20%7B%7D%20%3D%20%7B%20a%3A%207%20%7D%3B%0A%0Ax%20%3D%20'789'%3B%0A%0Aconsole.log(x)%3B%0A%0A%0Afunction%20printAll(list%3A%20Array%3C%7B%7D%3E)%20%7B%0A%20%20%20%20list.forEach(i%20%3D%3E%20console.log(i))%3B%0A%7D%0A%0AprintAll(%5B'avc'%2C%20123%2C%20true%2C%20null%5D)%3B). That's why the `object` type was introduced- it can't be assigned primitives.

Reply:
Consumers should be able to pass in primitives as a result though. E.g. a prompt dialog can return a boolean based on whether the user decided to accept something.